d3ploy/demand_driven_deployment_inst.py

```diff
@@ -219,17 +241,28 @@ def decision(self):
         """
         time = self.context.time
         for commod, proto_dict in self.commodity_dict.items():
-
             diff, supply, demand = self.calc_diff(commod, time)
             lib.record_time_series('calc_supply' + commod, self, supply)
             lib.record_time_series('calc_demand' + commod, self, demand)
-
             if diff < 0:
-                deploy_dict = solver.deploy_solver(
-                    self.commodity_supply, self.commodity_dict, commod, diff, time)
+                if self.installed_cap:
```

Comment:
These lines are unnecessarily complicated.
I recommend creating the following function:

```
def target():
  if self.installed_cap:
    return self.installed_capacity
  else:
    return self.commodity_supply
```

Then, you can turn all of this copy pasted stuff into just one line.

```
deploy_dict = solver.deploy_solver(self.target(), self. commodity_dict, commod, diff, time)
```

Reply:
I implemented this